src/AppInstallerCLICore/Workflows/ShellExecuteInstallerHandler.cpp

```diff
@@ -307,7 +347,142 @@ namespace AppInstaller::CLI::Workflow
+    {
+        if (!Settings::ExperimentalFeature::IsEnabled(Settings::ExperimentalFeature::Feature::WindowsFeature))
+        {
+            return;
+        }
+
+        const auto& rootDependencies = context.Get<Execution::Data::Installer>()->Dependencies;
+        DWORD result = 0;
+        bool force = context.Args.Contains(Execution::Args::Type::Force);
+        bool rebootRequired = false;
+
+        rootDependencies.ApplyToType(DependencyType::WindowsFeature, [&context, &result, &force, &rebootRequired](Dependency dependency)
+            {
+                if (FAILED(result) && !force || context.IsTerminated())
+                {
+                    return;
+                }
+
+                auto featureName = dependency.Id();
+                Utility::LocIndView locIndFeatureName{ featureName };
+
+                // Check if Windows Feature exists.
+#ifndef AICLI_DISABLE_TEST_HOOKS
+                auto doesFeatureExistResult = s_DoesWindowsFeatureExistResult_Override.has_value() ?
+                    s_DoesWindowsFeatureExistResult_Override.value() :
+                    DoesWindowsFeatureExist(context, featureName);
+#else
+                auto doesFeatureExistResult = DoesWindowsFeatureExist(context, featureName);
+#endif
 
+                if (!doesFeatureExistResult)
+                {
+                    // User aborted the operation.
+                    AICLI_TERMINATE_CONTEXT(E_ABORT);
+                }
+
+                auto doesFeatureExistExitCode = doesFeatureExistResult.value();
+                if (doesFeatureExistExitCode != ERROR_SUCCESS)
+                {
+                    AICLI_LOG(Core, Info, << "Windows Feature [" << featureName << "] does not exist");
+                    context.Reporter.Warn() << Resource::String::WindowsFeatureNotFound(locIndFeatureName) << std::endl;
+                    result = doesFeatureExistExitCode;
+                    return;
+                }
+
+                // Enable Windows Feature.
+                context.Reporter.Info() << Resource::String::EnablingWindowsFeature(locIndFeatureName) << std::endl;
+
+#ifndef AICLI_DISABLE_TEST_HOOKS
+                auto enableFeatureResult = s_EnableWindowsFeatureResult_Override.has_value() ?
+                    s_EnableWindowsFeatureResult_Override.value() :
+                    EnableWindowsFeature(context, featureName);
+#else
+                auto enableFeatureResult = EnableWindowsFeature(context, featureName);
+#endif
+                if (!enableFeatureResult)
+                {
+                    // User aborted the operation.
+                    AICLI_TERMINATE_CONTEXT(E_ABORT);
+                }
+
+                auto enableFeatureExitCode = enableFeatureResult.value();
+                if (enableFeatureExitCode == ERROR_SUCCESS_REBOOT_REQUIRED)
+                {
+                    AICLI_LOG(Core, Info, << "Reboot required for [" << featureName << "] with exit code: " << result);

```

Comment:
```suggestion
                    AICLI_LOG(Core, Info, << "Reboot required for [" << featureName << ']');

```
The result is not assigned, it's the value of last run